### PR TITLE
New GitHub action to merge autogenerated

### PR DESCRIPTION
**Pull request by pipo02mix**
### What this PR does / why we need it

The idea is to merge automatically the content autogenerated during the week. In order to bypass the branch protection I created an app like described [here](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/making-authenticated-api-requests-with-a-github-app-in-a-github-actions-workflow)

### Things to check/remember before submitting

- If you made content changes

    - Run `make dev` to render and proofread content changes locally.
    - Bump `last_review_date` in the front matter header if you reviewed the entire page.

### CODE REVIEW

File: .github/workflows/merge-autogenerated-content.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - cron: "0 10 * * FRI"
+  workflow_dispatch: {} # Allow manual triggering
+
+jobs:
+  merge-autogenerated-content:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Generate a token
+      id: generate-token
+      uses: actions/create-github-app-token@v1
```
Comment (marians) [inline .github/workflows/merge-autogenerated-content.yml:16]: Let's please pin the version for security

```suggestion
      uses: actions/create-github-app-token@7bfa3a4717ef143a604ee0a99d859b8886a96d00 # v1.9.3
```

File: .github/workflows/merge-autogenerated-content.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      with:
+        app-id: ${{ vars.AUTO_MERGE_DOCS_PRS_APP }}
+        private-key: ${{ secrets.AUTO_MERGE_DOCS_PRS_APP }}
+
+    - name: Checkout code
+      uses: actions/checkout@v2
```
Comment (marians) [inline .github/workflows/merge-autogenerated-content.yml:22]: Any reason not to use the latest? Also please pin to commit SHA.

```suggestion
      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 v4.1.2
```

File: .github/workflows/merge-autogenerated-content.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+    - name: Generate a token
+      id: generate-token
+      uses: actions/create-github-app-token@7bfa3a4717ef143a604ee0a99d859b8886a96d00 # v1.9.3
+      with:
+        app-id: ${{ vars.AUTO_MERGE_DOCS_PRS_APP }}
+        private-key: ${{ secrets.AUTO_MERGE_DOCS_PRS_APP }}
+
+    - name: Checkout code
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 v4.1.2
```
Comment (marians) [inline .github/workflows/merge-autogenerated-content.yml:22]: Forgot the comment sign. My bad!

```suggestion
      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
```

**Comment by pipo02mix** [comment]
@marians I guess I should remove [this one instead](https://github.com/giantswarm/docs/blob/main/.github/workflows/on-update-changes-approved.yml)

File: .github/workflows/merge-autogenerated-content.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        # List all open pull requests
+        PRS=$(gh pr list -S "open" -R ${{ github.repository }} -L 100 | awk '{print $1}')
+
+        for PR in $PRS
+        do
+          # Get the title of the current pull request
+          CURRENT_PR_TITLE=$(gh pr view $PR -R ${{ github.repository }} --json title -q '.title')
+
+          if [ "$CURRENT_PR_TITLE" == "$PR_TITLE" ]
```
Comment (AndiDog) [inline .github/workflows/merge-autogenerated-content.yml:36]: [minor]

```suggestion
          if [ "$CURRENT_PR_TITLE" = "$PR_TITLE" ]
```

to not be Bash-specific (not sure if it matters here)

File: src/content/overview/architecture/_index.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 title: Architecture
 description: Technical overview of the Giant Swarm platform components and how a customer's developer platform can use them successfully.
 weight: 20
+menu:
```
Comment (AndiDog) [inline src/content/overview/architecture/_index.md:5]: These changes don't seem to belong in this PR

**Review by AndiDog — COMMENTED** [review]
LGTM otherwise

File: src/content/overview/architecture/_index.md (hunk 5 above)
Comment (pipo02mix) [reply src/content/overview/architecture/_index.md:5]: yeah let me pull from main

**Review by AndiDog — APPROVED** [review]
